### PR TITLE
{bio}[foss/2023a] funannotate v1.8.17 + required dependencies

### PR DESCRIPTION
**Pull request by laraPPr**
(created using `eb --new-pr`)

- [x] https://github.com/easybuilders/easybuild-easyconfigs/pull/21569#issuecomment-2422312501
- [x] https://github.com/easybuilders/easybuild-easyconfigs/pull/21704

### CODE REVIEW

File: easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+builddependencies = [
+    ('Autotools', '20220317'),
+    ('CMake', '3.26.3'),
+]
+
+# for reference, list of dependencies in the container image used upstream:
+# https://github.com/trinityrnaseq/trinityrnaseq/blob/master/Docker/Dockerfile
+dependencies = [
+    ('Java', '11', '', SYSTEM),
+    ('ant', '1.10.12', '-Java-%(javaver)s', SYSTEM),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2023a.eb:38]: @laraPPr This should be bumped to `1.10.14`, to fix CI failure:

```
Found 2 variants of 'ant' dependency in easyconfigs using '2023a' toolchain generation
``

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2023a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2023a.eb:38]:
```suggestion
    ('ant', '1.10.14', '-Java-%(javaver)s', SYSTEM),

File: easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2023a.eb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# Homepage:     https://staff.flinders.edu.au/research/deep-thought
+#
+# Authors::     Robert Qiao <rob.qiao@flinders.edu.au>
+# License::     Custom
+#
+# Notes::
+##
+
+name = 'Trinity'
+version = '2.15.1'
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2023a.eb:12]: @laraPPr Any reason we're not using `2.15.2`?

Comment (laraPPr) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.15.1-foss-2023a.eb:12]: No, I must have overlooked it. Bumping it

**Comment by laraPPr** [comment]
@boegelbot please test @ generoso

**Comment by laraPPr** [comment]
`FAIL (build issue) GeneMark-ET-4.72-GCCcore-12.3.0.eb` 
I'm not sure why it failed on this one since it is not in this PR

**Comment by laraPPr** [comment]
@boegelbot please test @ jsc-zen3

**Comment by laraPPr** [comment]
Their is a problem with dependency https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/t/tbl2asn/tbl2asn-20230713-linux64.eb which is not installing with RPATH enabled
```
== 2024-10-25 10:50:32,996 run.py:260 INFO running cmd: ldd /data/gent/vo/000/gvo00002/vsc46128/easybuild/RHEL9/skylake-ib/software/tbl2asn/20230713-linux64/bin/tbl2asn
== 2024-10-25 10:50:33,004 easyblock.py:3180 WARNING Library libidn.so.11 not found for /data/gent/vo/000/gvo00002/vsc46128/easybuild/RHEL9/skylake-ib/software/tbl2asn/20230713-linux64/bin/tbl2asn
== 2024-10-25 10:50:33,318 build_log.py:171 ERROR EasyBuild crashed with an error (at easybuild/eb_job_skitty_P3.11_venv/lib/python3.11/site-packages/easybuild/tools/build_log.py:111 in caller_info): Sanity check failed: Library libidn.so.11 not found for /data/gent/vo/000/gvo00002/vsc46128/easybuild/RHEL9/skylake-ib/software/tbl2asn/20230713-linux64/bin/tbl2asn (at easybuild/eb_job_skitty_P3.11_venv/lib/python3.11/site-packages/easybuild/framework/easyblock.py: 3670 in _sanity_check_step)
```
Will try to do something similar as here https://github.com/easybuilders/easybuild-easyconfigs/pull/21673 as suggested in the slack channel.

**Comment by laraPPr** [comment]
Needs another tbl2asn for RPATH enabled builds
- [ ] https://github.com/easybuilders/easybuild-easyconfigs/pull/21756

File: easybuild/easyconfigs/t/Trinity/Trinity-2.15.2-foss-2023a.eb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+]
+
+builddependencies = [
+    ('Autotools', '20220317'),
+    ('CMake', '3.26.3'),
+]
+
+# for reference, list of dependencies in the container image used upstream:
+# https://github.com/trinityrnaseq/trinityrnaseq/blob/master/Docker/Dockerfile
+dependencies = [
```
Comment (hajgato) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.15.2-foss-2023a.eb:36]: DB_File.pm is missing. Need to add DB_File as dep `('DB_File', '1.859'),`

**Comment by hajgato** [comment]
Trinity needs DB_File as dep (`DB_File/1.859-GCCcore-12.3.0`)

File: easybuild/easyconfigs/t/Trinity/Trinity-2.15.2-foss-2023a.eb (hunk 3 above)
Comment (laraPPr) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.15.2-foss-2023a.eb:36]: I think it's ('DB', '18.1.40') that is missing DB was a dependency of  Perl-5.34.1-GCCcore-11.3.0.eb but is not a part of 
```
 ('Perl', '5.36.1'),
 ('Perl-bundle-CPAN', '5.36.1'),
```

File: easybuild/easyconfigs/t/Trinity/Trinity-2.15.2-foss-2023a.eb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    ('Java', '11', '', SYSTEM),
+    ('ant', '1.10.14', '-Java-%(javaver)s', SYSTEM),
+    ('picard', '2.25.1', '-Java-%(javaver)s', SYSTEM),
+    ('GATK', '4.3.0.0', '-Java-%(javaver)s'),
+    ('Perl', '5.36.1'),
+    ('Perl-bundle-CPAN', '5.36.1'),
+    ('DB', '18.1.40'), # for DB_File
```
Comment (laraPPr) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.15.2-foss-2023a.eb:43]:
```suggestion
    ('DB', '18.1.40'),  # for DB_File
```

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 15 out of 15 (15 easyconfigs in total)
node3103.skitty.os - Linux RHEL 9.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.9.18
See https://gist.github.com/boegel/a35f51fe2dc49512f66dc60a2237ab9b for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @laraPPr!

**Pull request merged**